Wrappers/Python/cil/optimisation/algorithms/Algorithm.py

```diff
@@ -128,20 +128,25 @@ def __next__(self):
             
 
     def update_previous_solution(self):
-        '''Update the previous solution with the current one
+        """ Update the previous solution with the current one
         
         The concrete algorithm calls update_previous_solution. Normally this would 
         entail the swapping of pointers:
 
-        tmp = self.x_old
-        self.x_old = self.x
-        self.x = tmp 
-        '''
+        .. highlight:: python
+        .. code-block:: python
+
+            tmp = self.x_old
+            self.x_old = self.x
+            self.x = tmp 
+        
+
+        """
         pass
         
     def get_output(self):
         '''Returns the solution found'''
-        return self.x
+        return self.x_old
```

Comment:
I mean, that the current solution is in `self.x` normally. Now, in ISTA/FISTA and PDHG/SPDHG it is in `self.x_old` because we swap the pointers at the end of each iteration.

But this is a special thing of these algorithms. So it needs to be in these that `get_output` returns `self.x_old` not in the main class.

Reply:
`get_output` added to `ISTA` algorithm. `FISTA` will inherit it from `ISTA`.

However, for the algorithms that we have atm in CIL, we can have both `get_output` and `update_previous_solution` in the base class.